tiledb/sm/array/array_directory.cc

```diff
@@ -324,6 +446,93 @@ tuple<Status, optional<URI>> ArrayDirectory::load_fragment_metadata_dir_uris() {
   return {Status::Ok(), std::move(fragment_meta_uri.value())};
 }
 
+tuple<
+    Status,
+    optional<std::vector<URI>>,
+    optional<std::unordered_set<std::string>>>
+ArrayDirectory::load_consolidated_commit_uris(
+    const std::vector<URI>& commits_dir_uris) {
+  // Load the commit URIs to ignore
+  std::unordered_set<std::string> ignore_set;
+  for (auto& uri : commits_dir_uris) {
+    if (stdx::string::ends_with(
+            uri.to_string(), constants::ignore_file_suffix)) {
+      uint64_t size = 0;
+      RETURN_NOT_OK_TUPLE(vfs_->file_size(uri, &size), nullopt, nullopt);
+      std::string names;
+      names.resize(size);
+      RETURN_NOT_OK_TUPLE(
+          vfs_->read(uri, 0, &names[0], size), nullopt, nullopt);
+      std::stringstream ss(names);
+      for (std::string uri_str; std::getline(ss, uri_str);) {
+        ignore_set.emplace(uri_str);
+      }
+    }
+  }
+
+  // Load all commit URIs
+  std::unordered_set<std::string> uris_set;
+  std::vector<std::pair<URI, std::string>> meta_files;
+  for (uint64_t i = 0; i < commits_dir_uris.size(); i++) {
+    auto& uri = commits_dir_uris[i];
+    if (stdx::string::ends_with(
+            uri.to_string(), constants::commits_file_suffix)) {
+      uint64_t size = 0;
+      RETURN_NOT_OK_TUPLE(vfs_->file_size(uri, &size), nullopt, nullopt);
+      meta_files.emplace_back(uri, std::string());
+
+      auto& names = meta_files.back().second;
+      names.resize(size);
+      RETURN_NOT_OK_TUPLE(
+          vfs_->read(uri, 0, &names[0], size), nullopt, nullopt);
+      std::stringstream ss(names);
+      for (std::string uri_str; std::getline(ss, uri_str);) {
+        if (ignore_set.count(uri_str) == 0) {
+          uris_set.emplace(uri_.to_string() + uri_str);
+        }
+      }
+    }
+  }
+
+  // Make a sorted vector from the set
+  std::vector<URI> uris;
+  uris.reserve(uris_set.size());
+  for (auto& uri : uris_set) {
+    uris.emplace_back(uri);
+  }
+  std::sort(uris.begin(), uris.end());
```

Comment:
Is this unnecessary? I think the set iterator already visits the URIs in sorted order. If not, perhaps making a sorted map passing the URI comparator is a better idea here?

Reply:
I think so. The unordered_set gets used in multiple functions later. And changing it to a set makes the count function logarithmic instead of constant time. So this might be better?